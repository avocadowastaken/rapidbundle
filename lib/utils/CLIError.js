export class CLIError extends Error {
  /** @param {string} message */
  constructor(message) {
    super(message);
    this.name = "CLIError";
  }
}
