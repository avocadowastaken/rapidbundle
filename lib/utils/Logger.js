"use strict";

const ora = require("ora");
const prettyMS = require("pretty-ms");
const CLIError = require("./CLIError");

module.exports = class Logger {
  /** @param {NodeJS.WritableStream} stream */
  constructor(stream) {
    /**
     * @type {NodeJS.WritableStream}
     * @readonly
     * @protected
     */
    this.stream = stream;
  }

  /**
   * @template {unknown} T
   * @param {string} name
   * @param {(spinner: import("ora").Ora) => Promise<T>} fn
   * @returns {Promise<T>}
   */
  async time(name, fn) {
    const spinner = ora({ stream: this.stream }).start(name);

    try {
      const startedAt = Date.now();
      const result = await fn(spinner);
      spinner.succeed(`${name} (${prettyMS(Date.now() - startedAt)})`);
      return result;
    } catch (e) {
      if (e instanceof CLIError) {
        spinner.fail(e.message);
      } else if (e instanceof Error) {
        spinner.fail(e.stack);
      }

      throw e;
    }
  }
};
