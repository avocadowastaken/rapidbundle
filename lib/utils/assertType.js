"use strict";

const CLIError = require("./CLIError");

/**
 * @template {unknown} U
 * @template {U} V
 * @param {import("typanion").StrictTest<U, V>} isValid
 * @param {U} value
 * @param {string} [message]
 * @returns {asserts value is V}
 */
module.exports = function assertType(isValid, value, message = "Invalid type") {
  /** @type {string[]} */
  const errors = [];

  if (!isValid(value, { errors })) {
    throw new CLIError(
      `${message}:\n${errors.map((error) => `  ${error}`).join("\n")}`
    );
  }
};
