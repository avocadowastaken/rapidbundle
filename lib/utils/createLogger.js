"use strict";

const debug = require("debug");

/**
 * @param {...string} namespace
 * @returns {import("debug").Debugger}
 */
module.exports = function createLogger(...namespace) {
  const logger = debug(["rapidbundle", ...namespace].join(":"));
  logger.log = console.log.bind(console);
  return logger;
};
