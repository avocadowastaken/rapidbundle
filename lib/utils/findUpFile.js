const path = require("path");
const findUp = require("find-up");

/**
 * @param {string} input
 * @param {string} fileName
 * @returns {Promise<string | undefined>}
 */

module.exports = async function findUpFile(input, fileName) {
  if (!path.isAbsolute(input)) input = path.resolve(input);
  if (path.basename(input) === fileName) return input;
  return findUp(fileName, { type: "file", cwd: input });
};
