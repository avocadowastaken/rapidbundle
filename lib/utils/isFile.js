import { promises as fs } from "fs";

/**
 * @param {string} input
 * @returns {Promise<boolean>}
 */
export async function isFile(input) {
  try {
    const stat = await fs.stat(input);
    return stat.isFile();
  } catch {
    return false;
  }
}
