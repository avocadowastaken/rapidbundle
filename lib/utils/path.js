import path from "path";

/**
 * @param {string} rootDir
 * @param {string} outputFile
 */
export function resolveEntry(rootDir, outputFile) {
  const srcDir = path.join(rootDir, "src");
  const distDir = path.join(rootDir, "dist");
  const outputDir = path.dirname(outputFile);

  const entryDir = path.relative(distDir, outputDir);
  const entryBaseName = path.basename(outputFile, path.extname(outputFile));

  return path.join(srcDir, entryDir, entryBaseName);
}

/**
 * @param {string} rootDir
 * @param {string} input
 * @returns {string}
 */
export function formatRelativePath(rootDir, input) {
  return `./${path.relative(rootDir, input).replace(/\\/g, "/")}`;
}
