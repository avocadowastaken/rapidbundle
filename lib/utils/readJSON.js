const { promises: fs } = require("fs");
const StandardError = require("./StandardError");

/** @param {string} filePath */
module.exports = async function readJSON(filePath) {
  try {
    const json = await fs.readFile(filePath, "utf-8");
    return JSON.parse(json);
  } catch {
    throw new StandardError(`Failed to parse ${filePath} to json`);
  }
};
