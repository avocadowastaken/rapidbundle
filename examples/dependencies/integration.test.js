import { runIntegrationTest } from "../../test/utils/runTest";

runIntegrationTest(import.meta.url);
