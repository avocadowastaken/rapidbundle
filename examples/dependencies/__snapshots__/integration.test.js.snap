// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dependencies: dist 1`] = `
Array [
  "index.cjs",
  "index.js",
]
`;

exports[`dependencies: dist/index.cjs 1`] = `
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __markAsModule = (target) => __defProp(target, "__esModule", { value: true });
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  __markAsModule(target);
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __reExport = (target, module2, desc) => {
  if (module2 && typeof module2 === "object" || typeof module2 === "function") {
    for (let key of __getOwnPropNames(module2))
      if (!__hasOwnProp.call(target, key) && key !== "default")
        __defProp(target, key, { get: () => module2[key], enumerable: !(desc = __getOwnPropDesc(module2, key)) || desc.enumerable });
  }
  return target;
};
var __toModule = (module2) => {
  return __reExport(__markAsModule(__defProp(module2 != null ? __create(__getProtoOf(module2)) : {}, "default", module2 && module2.__esModule && "default" in module2 ? { get: () => module2.default, enumerable: true } : { value: module2, enumerable: true })), module2);
};

// src/index.ts
__export(exports, {
  hello: () => hello
});

// ../../node_modules/strip-ansi/index.js
var import_ansi_regex = __toModule(require("ansi-regex"));
function stripAnsi(string) {
  if (typeof string !== "string") {
    throw new TypeError(\`Expected a \\\`string\\\`, got \\\`\${typeof string}\\\`\`);
  }
  return string.replace((0, import_ansi_regex.default)(), "");
}
__name(stripAnsi, "stripAnsi");

// src/hello.ts
function hello(name) {
  return \`Hello, \${stripAnsi(name)}\`;
}
__name(hello, "hello");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  hello
});

`;

exports[`dependencies: dist/index.js 1`] = `
// ../../node_modules/strip-ansi/index.js
import ansiRegex from "ansi-regex";
function stripAnsi(string) {
  if (typeof string !== "string") {
    throw new TypeError(\`Expected a \\\`string\\\`, got \\\`\${typeof string}\\\`\`);
  }
  return string.replace(ansiRegex(), "");
}

// src/hello.ts
function hello(name) {
  return \`Hello, \${stripAnsi(name)}\`;
}
export {
  hello
};

`;

exports[`dependencies: logs 1`] = `
[STARTED] Resolving build manifests
[DATA] Checking 'action.yml'
[DATA] Checking 'package.json'
[SUCCESS] Resolving build manifests
[STARTED] Run preparations
[DATA] Removing 'dist' directory
[SUCCESS] Run preparations
[STARTED] Making bundle from 'package.json'
[STARTED] Parsing 'package.json'
[DATA] Using ".dependencies" as external: ansi-regex
[SUCCESS] Parsing 'package.json'
[STARTED] Making '.main' entry bundle
[DATA] Setting entry point: ./src/index
[DATA] Setting output file: ./dist/index.cjs
[SUCCESS] Making '.main' entry bundle
[STARTED] Making '.module' entry bundle
[DATA] Setting entry point: ./src/index
[DATA] Setting output file: ./dist/index.js
[DATA] Setting build target: chrome91, edge92, firefox78, ios12.2, safari13.1
[SUCCESS] Making '.module' entry bundle
[SUCCESS] Making bundle from 'package.json'
`;
